### PR TITLE
Reimplemented ProjectViewExtension and minor fixes (for MPS 2024.1 a.k.a "master")

### PR DESCRIPTION
**Pull request by fisakov**
This changeset comprises a reimplementation of ProjectViewExtension from "shadow models" on top of the new ProjectPane API, as well as a few minor fixes that make the project build with the latest master version of MPS. 

See [this article](https://youtrack.jetbrains.com/articles/MPS-A-216170516/ProjectPane-implementation-on-top-of-ProjectViewTree) for details. 

Note, that this pull request, although made on top of `mps/2023.3` branch, is to be merged in a branch that should correspond to the next available version of MPS, which is going to include the changes necessary for the above fixes to work, and only after these changes are merged into our (MPS) master branch. 

In short, things should happen in this order (ideally): 

1. ✅ we merge the changes that implement the new ProjectPane and test with a temporary branch of MPS-extensions (the one referred to in this pull request)
2. you introduce a new branch to track fixes necessary for compatibility with MPS 2024.1 (e.g. mps/2024.1)
3. we switch to that branch for our integration builds
4. MPS 2024.1 is released containing the new ProjectPane and MPS-extensions/mps/2024.1 is up-to-date with it.

### CODE REVIEW

**Comment by fisakov** [comment]
Please use this branch from MPS repo to test: 2024.1_projectPane

**Comment by sergej-koscejev** [comment]
The plan is to move MPS-extensions master to MPS master (prereleases) and then merge this PR.
